File: softtools/test/transactor/mongostrategy.cpp
#include <QDebug>
#include "mongostrategy.h"

MongoStrategy :: MongoStrategy()  
  : StorageStrategy()
{}

MongoStrategy :: ~MongoStrategy()
{}

bool MongoStrategy :: store (double value)
{
  qDebug() << "storing" << value;
  return true;
}

File: softtools/test/transactor/f90/strategy/mongostrategy.f90
module MongoStrategy
  use StorageStrategy, only : IStorageStrategy
  implicit none
  private

  type, extends(IStorageStrategy), public :: TMongoStrategy
   contains
     procedure :: create
  end type TMongoStrategy

contains
  subroutine create (this)
    class (TMongoStrategy) :: this
    print *, 'hello from mongo'

  end subroutine create

end module MongoStrategy
